File: atividade-4/018_frequencia/018_frequencia.hs
module Main where

-- Encontrar o total ocorrencias de n em um vetor xs

frequencia :: Int -> [Int] -> Int
frequencia n [] = 0
frequencia n (x:xs) 
          | x == n = 1 + (frequencia n xs)
          | otherwise = frequencia n xs

main = do
  a <- readLn :: IO Int
  b <- readLn :: IO [Int]
  print $ frequencia a b

File: atividade-4/018_frequencia/018_frequencia.js
//Encontrar o total ocorrencias de n em um vetor xs

const list = [1, 1, 2, 6, 6, 6, 7, 7, 7];
const number = 6;

const frequencia = (n, xs) => {
  if (xs.length === 0) return 0;

  const x = xs.shift();

  if (n === x) return 1 + frequencia(n, xs);

  return frequencia(n, xs);
};

console.log(frequencia(number, list));
